### PR TITLE
Ipfs object put from stdin

### PR DESCRIPTION
**Pull request by cryptix**
follow up to #811

### CODE REVIEW

File: core/commands/object.go [hunk 1]
```diff
@@ -274,16 +274,18 @@ It reads from stdin, and the output is a base58 encoded multihash.
...
 `,
 	},
 
 	Arguments: []cmds.Argument{
-		cmds.FileArg("data", true, false, "Data to be stored as a DAG object"),
-		cmds.StringArg("encoding", true, false, "Encoding type of <data>, either \"protobuf\" or \"json\""),
+		cmds.FileArg("data", true, false, "Data to be stored as a DAG object").EnableStdin(),
+	},
+	Options: []cmds.Option{
+		cmds.StringOption("inputenc", "Encoding type of input data, either \"protobuf\" or \"json\""),
```
Comment (jbenet) [inline core/commands/object.go]: was it not possible to read the already standard `--enc` flag? does the commands lib hide it away?

Comment (cryptix) [inline core/commands/object.go]: I thought it would be nicer to have two flags, so you can define input and output encoding independently.

Comment (jbenet) [inline core/commands/object.go]: ah, okay :+1:

**Comment by jbenet** [comment]
Awesome, LGTM!

**Pull request merged**